xdsl/dialects/experimental/dmp.py

```diff
@@ -426,6 +427,160 @@ def print_parameters(self, printer: Printer) -> None:
         printer.print_string(">")
 
 
+class DomainDecompositionStrategy(ParametrizedAttribute, ABC):
+    def calc_resize(self, shape: tuple[int, ...]) -> tuple[int, ...]:
+        raise NotImplementedError("SlicingStrategy must implement calc_resize!")
+
+    def halo_exchange_defs(self, shape: ShapeAttr) -> Iterable[ExchangeDeclarationAttr]:
+        raise NotImplementedError("SlicingStrategy must implement halo_exchange_defs!")
+
+    def comm_layout(self) -> RankTopoAttr:
+        raise NotImplementedError("SlicingStrategy must implement comm_count!")
+
+
+@irdl_attr_definition
+class GridSlice2dAttr(DomainDecompositionStrategy):
+    """
+    Takes a grid with two or more dimensions, slices it along the first two into equally
+    sized segments.
+    """
+
+    name = "dmp.grid_slice_2d"
+
+    topology: ParameterDef[RankTopoAttr]
+
+    diagonals: ParameterDef[builtin.BoolAttr]
```

Comment:
Would this make sense as an optional unitattr? the IR would be a bit more self-explanatory than just `false`

Reply:
How?
https://xdsl.zulipchat.com/#narrow/stream/364199-Beginners/topic/Optional.20attribute.20parameter.20def

Reply:
I'm also not trying to perfect the syntax just yet here!

Reply:
OK then please disregard :)